### PR TITLE
Improve Editing and Navigating shortcuts with Ctrl/Meta

### PR DESCRIPTION
**Pull request by MayurSMahajan**
Solves #75 #1825 

Adds the following functionalities:

- [x] Ctrl + Backspace to delete a word.
- [x] Ctrl + Alt + Arrow Key Left to select the left word (left to the cursor).
- [x] Ctrl + Alt + Arrow Key Right to select the right word (right to the cursor).

### CODE REVIEW

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart [hunk 1]
```diff
@@ -51,14 +51,25 @@ List<ShortcutEvent> builtInShortcutEvents = [
   ),
   ShortcutEvent(
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart:52]: I think there is no need to add more built-in commands for word selection now. I prefer to provide a shortcut commands panel if the users want to modify them.

```dart
   key: 'Cursor down select',
    command: 'shift+alt+arrow left,meta+alt+arrow left',
    windowsCommand: 'shift+alt+arrow left,ctrl+alt+arrow left',
    linuxCommand: 'shift+alt+arrow left,ctrl+alt+arrow left',
    handler: cursorLeftWordSelect,
  ),
  ShortcutEvent(
    key: 'Cursor down select',
    command: 'shift+alt+arrow right,ctrl+alt+arrow right',
    windowsCommand: 'shift+alt+arrow right,ctrl+alt+arrow right',
    linuxCommand: 'shift+alt+arrow right,ctrl+alt+arrow right',
    handler: cursorRightWordSelect,
  ),
```

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/arrow_keys_handler.dart [hunk 2]
```diff
@@ -322,6 +322,31 @@ ShortcutEventHandler cursorRightWordSelect = (editorState, event) {
   return KeyEventResult.handled;
 };
 
+ShortcutEventHandler cursorLeftWordDelete = (editorState, event) {
+  final nodes = editorState.service.selectionService.currentSelectedNodes;
+  final selection = editorState.service.selectionService.currentSelection.value;
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/arrow_keys_handler.dart:327]: if the `nodes.whereType<TextNode>()` returns an empty list, it will throw an error.

```dart
ShortcutEventHandler cursorLeftWordDelete = (editorState, event) {
  final textNodes = editorState.service.selectionService.currentSelectedNodes
      .whereType<TextNode>();
  final selection = editorState.service.selectionService.currentSelection.value;
  if (textNodes.isEmpty || selection == null) {
    return KeyEventResult.ignored;
  }

  final textNode = textNodes.first;

  final startOfWord =
      selection.end.goLeft(editorState, selectionRange: _SelectionRange.word);
  if (startOfWord == null) {
    return KeyEventResult.ignored;
  }

  final transaction = editorState.transaction;
  transaction.deleteText(
    textNode,
    startOfWord.offset,
    selection.end.offset - startOfWord.offset,
  );
  editorState.apply(transaction);

  return KeyEventResult.handled;
};
```

File: frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart [hunk 3]
```diff
@@ -466,6 +466,58 @@ void main() async {
       ),
     );
   });
+
+  testWidgets('Presses ctrl + backspace to delete a word', (tester) async {
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart:470]: I suggest adding more tests to cover the edge cases.

Like
A.
1. the before text: welcome to app|flowy.
2. pressing ctrl+backspace
3. the after text: welcome to app.

B.
1. the before text: |welcome to appflowy.
2. pressing ctrl+backspace
3. nothing happens.

File: frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart [hunk 4]
```diff
@@ -466,6 +466,58 @@ void main() async {
...
+
+    await editor.pressLogicKey(
+      LogicalKeyboardKey.backspace,
+      isControlPressed: true,
+    );
+
+    //fetching all the text that is still on the editor.
+    var nodes =
+        editor.editorState.service.selectionService.currentSelectedNodes;
+    var textNode = nodes.whereType<TextNode>().toList(growable: false).first;
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart:487]: No need to use `.toList()`

File: frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart [hunk 5]
```diff
@@ -466,6 +466,58 @@ void main() async {
...
+
+    for (var i = 0; i < words.length; i++) {
+      await editor.pressLogicKey(
+        LogicalKeyboardKey.backspace,
+        isControlPressed: true,
+      );
+    }
+
+    nodes = editor.editorState.service.selectionService.currentSelectedNodes;
+    textNode = nodes.whereType<TextNode>().toList(growable: false).first;
```
Comment (LucasXu0) [inline frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart:515]: same as the previous one.

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart (hunk 1 above)
Comment (MayurSMahajan) [reply frontend/app_flowy/packages/appflowy_editor/lib/src/service/shortcut_event/built_in_shortcut_events.dart:52]: Sure

File: frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/arrow_keys_handler.dart (hunk 2 above)
Comment (MayurSMahajan) [reply frontend/app_flowy/packages/appflowy_editor/lib/src/service/internal_key_event_handlers/arrow_keys_handler.dart:327]: On it

File: frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart (hunk 3 above)
Comment (MayurSMahajan) [reply frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart:470]: Sure. But in Case A. actually, we expect the after text to be 
` welcome to flowy.`

File: frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart (hunk 5 above)
Comment (MayurSMahajan) [reply frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart:515]: Right

File: frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart (hunk 4 above)
Comment (MayurSMahajan) [reply frontend/app_flowy/packages/appflowy_editor/test/service/internal_key_event_handlers/arrow_keys_handler_test.dart:487]: Right

**Comment by LucasXu0** [comment]
LGTM.
